packages/cli/src/utils/local.js

```diff
@@ -2,8 +2,231 @@ const _ = require('lodash');
+    let finalHost = targetHost;
+    if (targetPort && targetPort !== '443') {
+      finalHost += `:${targetPort}`;
+    }
+    const combinedPath = `${parsedRelayUrl.pathname}/${finalHost}${options.path}`;
+
+    // 5. Build final options for the relay request
+    const relayedOptions = {
+      protocol: parsedRelayUrl.protocol,
+      hostname: relayHost,
+      port: relayPort,
+      path: combinedPath,
+      method: options.method,
+      headers: {
+        ...options.headers,
+        ...relayHeaders,
+      },
+    };
+
+    // 6. Make the relay request
+    const relayReq = relayRequestFunc(relayedOptions, callback);
+    return relayReq;
+  };
+}
+
+/**
+ * Wraps a fetch function so that all requests get relayed to a specified relay URL.
+ * The final relay URL includes: relayUrl + "/" + originalHost + originalPath
+ *
+ * @param {Function} fetchFunc - The original fetch function (e.g., global.fetch).
+ * @param {string} relayUrl - The base URL to which we relay. (e.g. 'https://my-relay.test')
+ * @param {Object} relayHeaders - Extra headers to add to each request sent to the relay.
+ * @returns {Function} A function with the same signature as `fetch(url, options)`.
+ *
+ * Usage:
+ *   const wrappedFetch = wrapFetchWithRelay(
+ *     fetch,
+ *     'https://my-relay.test',
+ *     { 'X-Relayed-By': 'MyRelayProxy' },
+ *   );
+ *
+ *   // Now when you do:
+ *   //   wrappedFetch('https://example.com/api/user?id=123', { method: 'POST' });
+ *   // it actually sends a request to:
+ *   //   https://my-relay.test/example.com/api/user?id=123
+ *   // with "X-Relayed-By" header included.
+ */
+function wrapFetchWithRelay(fetchFunc, relayUrl, relayHeaders) {
+  const parsedRelayUrl = new URL(relayUrl);
+
+  return async function wrappedFetch(originalUrl, originalOptions = {}) {
+    // Attempt to parse the originalUrl as an absolute URL
+    const parsedOriginalUrl = new URL(originalUrl);
+
+    // Build the portion that includes the original host (and port if present)
+    let host = parsedOriginalUrl.hostname;
+    // If there's a port that isn't 443 (for HTTPS) or 80 (for HTTP), append it
+    // (Adjust to your preferences; here we loosely check for 443 only.)
+    if (parsedOriginalUrl.port && parsedOriginalUrl.port.toString() !== '443') {
+      host += `:${parsedOriginalUrl.port}`;
+    }
+
+    const isAlreadyRelay =
+      parsedOriginalUrl.hostname === parsedRelayUrl.hostname &&
+      parsedOriginalUrl.port === parsedRelayUrl.port;
+    if (isAlreadyRelay) {
+      // Just call the original fetch function; do *not* re-relay
+      return fetchFunc(originalUrl, originalOptions);
+    }
+
+    // Combine the relay's pathname with the "host + path" from the original
+    // For example: relayUrl = http://my-relay.test
+    //   => parsedRelayUrl.pathname might be '/'
+    //   => combinedPath = '/example.com:8080/some/path'
+    const combinedPath = `${parsedRelayUrl.pathname}/${host}${parsedOriginalUrl.pathname}`;
```

Comment:
```suggestion
    const combinedPath = `${parsedRelayUrl.pathname}/${host}${parsedOriginalUrl.pathname}`.replaceAll('lcurly-', '{{').replaceAll('-rcurly', '}}');
```

Wondering if this is necessary just like it is done here https://github.com/zapier/zapier-platform/pull/968/files#diff-7640b561b92d724ea2ef660a2e675d5f917e95b5ebde5490430cfc2fdddc46e9R104-R106 for `wrapHttpRequestFuncWithRelay`

Reply:
Argh, I can see you have explained why it is not necessary here as `node-fetch` throws error on `{` and `}`.